src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -1161,6 +1162,77 @@ public static void Fill<T>(ref T refData, nuint numElements, T value)
             return -1; // not found
         }
 
+        public static int IndexOfAnyExcept<T>(ref T searchSpace, T value0, int length)
+        {
+            Debug.Assert(length >= 0, "Expected non-negative length");
+
+            for (int i = 0; i < length; i++)
+            {
+                if (!EqualityComparer<T>.Default.Equals(Unsafe.Add(ref searchSpace, i), value0))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        public static int IndexOfAnyExceptValueType<T>(ref T searchSpace, T value0, int length) where T : struct, IEquatable<T>
+        {
+            Debug.Assert(length >= 0, "Expected non-negative length");
+            Debug.Assert(value0 is byte or short or int or long, "Expected caller to normalize to one of these types");
+
+            if (!Vector128.IsHardwareAccelerated || length < Vector128<T>.Count)
+            {
+                for (int i = 0; i < length; i++)
+                {
+                    if (!Unsafe.Add(ref searchSpace, i).Equals(value0))
+                    {
+                        return i;
+                    }
+                }
+            }
+            else
```

Comment:
Have you considered adding `Vector256` code path here?

Reply:
Yeah. I wanted to get something in. If I can find time while on vk this week I'd like to vectorize all the other overloads, measure adding a 256 path, and look at consuming these APIs from Regex, in that order. Unlikely I'll get to all that, though, by Monday.